addons/mail/models/mail_template.py

```diff
@@ -29,6 +29,10 @@ def default_get(self, fields):
             res['model_id'] = self.env['ir.model']._get(res.pop('model')).id
         return res
 
+    def _get_non_abstract_models_domain(self):
+        non_abstract_models = self.env['ir.model'].search([]).filtered(lambda m: not self.env[m.model]._abstract)
+        return [('id', 'in', non_abstract_models.ids)]
```

Comment:
probably need sudo for ir.model read?
And a negative domain is probably better as there are much fewer abstract models than concrete.

Also note you could do all this with just the registry and save a search (downside is the domain ends up more complex since you only have names). Your call.
```suggestion
        registry = self.env.registry
        abstract_models = [model for model in registry if registry[model]._abstract]
        return [('model', 'not in', abstract_models)]
```

Reply:
Hello @reth-odoo 

This one looks neater than a search, hence changing code to this.
About validation check, it is already present. I was asking to just do filtering :sweat_smile: 

Thanks :smile: